core/src/main/java/org/apache/accumulo/core/fate/user/UserFateStore.java

```diff
@@ -123,104 +123,54 @@ public FateId create() {
-                throw new IllegalStateException("Unexpected column seen: " + colf + ":" + colq);
-            }
-          }
+  @Override
+  public boolean seedTransaction(String txName, FateId fateId, Repo<T> repo, boolean autoCleanUp) {
+    Supplier<FateMutator<T>> mutatorFactory =
+        () -> newMutator(fateId).requireStatus(TStatus.NEW).requireUnreserved().requireAbsentKey();
+    return seedTransaction(mutatorFactory, fateId.canonical(), txName, repo, autoCleanUp);
+  }
 
-          if (statusSeen == TStatus.NEW) {
-            verifyFateKey(fateId, fateKeySeen, fateKey);
-            // This will be the case if the mutation status is REJECTED but the mutation was written
-            if (reservationSeen.isPresent() && reservationSeen.orElseThrow().equals(reservation)) {
-              txStore = Optional.of(new FateTxStoreImpl(fateId, reservation));
-            } else if (reservationSeen.isEmpty()) {
-              // NEW/unseeded transaction and not reserved, so we can allow it to be reserved
-              // we tryReserve() since another thread may have reserved it since the scan
-              txStore = tryReserve(fateId);
-              // the status was known before reserving to be NEW,
-              // however it could change so check after reserving to avoid race conditions.
-              var statusAfterReserve =
-                  txStore.map(ReadOnlyFateTxStore::getStatus).orElse(TStatus.UNKNOWN);
-              if (statusAfterReserve != TStatus.NEW) {
-                txStore.ifPresent(txs -> txs.unreserve(Duration.ZERO));
-                txStore = Optional.empty();
-              }
-            }
-          } else {
-            log.trace(
-                "fate id {} tstatus {} fate key {} is reserved {} "
-                    + "has already been seeded with work (non-NEW status)",
-                fateId, statusSeen, fateKeySeen.orElse(null), reservationSeen.isPresent());
-          }
-        } catch (TableNotFoundException e) {
-          throw new IllegalStateException(tableName + " not found!", e);
-        }
-        break;
-      default:
-        throw new IllegalStateException("Unknown or unexpected status " + status);
+  private boolean seedTransaction(Supplier<FateMutator<T>> mutatorFactory, String logId,
+      String txName, Repo<T> repo, boolean autoCleanUp) {
+    int maxAttempts = 5;
+    for (int attempt = 0; attempt < maxAttempts; attempt++) {
+      var mutator = mutatorFactory.get();
+      mutator =
+          mutator.putName(serializeTxInfo(txName)).putRepo(1, repo).putStatus(TStatus.SUBMITTED);
+      if (autoCleanUp) {
+        mutator = mutator.putAutoClean(serializeTxInfo(autoCleanUp));
+      }
+      var status = mutator.tryMutate();
+      if (status == FateMutator.Status.ACCEPTED) {
+        // signal to the super class that a new fate transaction was seeded and is ready to run
+        seededTx();
+        return true;
+      } else if (status == FateMutator.Status.REJECTED) {
```

Comment:
Good point.  Looking over the way things works end to end if a lot of effort is made here to determine what happened with unknown then the calling code will not really care (because it just logs the result).  However the way the functions are structured that is not obvious and if future calling code did something besides just logging the result then it would not be correct in this corner case.  I will try to look at the end to end structure and clean this up to either handle that corner case or make the functions reflect a best effort impl.

Reply:
Looking at this some more I will make it try to determine what happened.  Even thought its just logging, having accurate log message is important for debugging.

Reply:
Tried to handle the unknown case and realized its impossible to determine what happened in this case.  Prior to these changes a reservation was acquired and that has unique uuid which can be used to determine if the reservation was acquired when unknown is seen.   Now that this code is using a single conditional mutation, no reservation is required.  The reservation is only required for updates that happen over multiple conditional mutations.

So since there is no way to know what happened, I adjusted the logging and code to reflect this in 89b70da

Reply:
Ah, yeah you're right...
The new comment for the UNKNOWN case looks good. The new mention of "*possibly* unable to seed" in FateLogger looks good.